### PR TITLE
test(e2e): add e2e tests for table component

### PR DESCRIPTION
**Pull request by nikvalor**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#5909

#### Please provide a brief summary of this pull request.
add e2e tests for table(core) component

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: e2e/wdio/core/tests/table.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+            saveElementScreenshot(tableCustomColumnsExample + table, 'table-custom-columns-example-' + getImageTagBrowserPlatform(),
+                tablePage.getScreenshotFolder());
+            expect(checkElementScreenshot(tableCustomColumnsExample + table, 'table-custom-columns-example-' + getImageTagBrowserPlatform(),
+                tablePage.getScreenshotFolder())).toBeLessThan(5, `element item state mismatch`);
+        });
+
+        it('should check search work correctly', () => {
+            scrollIntoView(tableCustomColumnsExample);
+            click(tableCustomColumnsExample + button);
+            setValue(dialogContent + inputGroup, 'date');
```
Comment (maxym-valor) [inline e2e/wdio/core/tests/table.e2e-spec.ts:101]: use your test variable dateTestText instead just string 'date', will be easier to read code wthout needed to find this variable in fixtures

Comment (maxym-valor) [inline e2e/wdio/core/tests/table.e2e-spec.ts:101]: use your test variable dateTestText instead just string 'date', will be easier to read code wthout needed to find this variable in fixtures

**Review by maxym-valor — CHANGES_REQUESTED** [review]

File: e2e/wdio/core/tests/table.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    });
+
+    describe('Check Table With Semantic Row Highlighting example', function() {
+
+        it('should check that checkbox work correctly', () => {
+            scrollIntoView(tableSemanticExample);
+            click(tableSemanticExample + markAllCheckboxes);
+            const checkboxLength = getElementArrayLength(tableSemanticExample + tableRow);
+            for (let i = 0; i < checkboxLength; i++) {
+                expect(getAttributeByName(tableSemanticExample + tableRow, 'aria-selected', i)).toBe('true');
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/table.e2e-spec.ts:130]: please add message for failure

File: e2e/wdio/core/tests/table.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                expect(isElementClickable(tablePopinExample + clickableTableRow, i)).toBe(true, `element with index ${i} not clickable`);
+            }
+        });
+
+        it('should check that checkbox work correctly', () => {
+            scrollIntoView(tablePopinExample);
+            click(tablePopinExample + markAllCheckboxes);
+            const checkboxLength = getElementArrayLength(tablePopinExample + tableRow);
+            for (let i = 20; i < checkboxLength; i++) {
+                expect(getAttributeByName(tablePopinExample + tableRow, 'aria-selected', i)).toBe('true');
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/table.e2e-spec.ts:150]: please add message for failure

File: e2e/wdio/core/tests/table.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+            for (let i = 0; i < tableRowLength; i++) {
+                expect(isElementClickable(tablePopinExample + clickableTableRow, i)).toBe(true, `element with index ${i} not clickable`);
+            }
+        });
+
+        it('should check that checkbox work correctly', () => {
+            scrollIntoView(tablePopinExample);
+            click(tablePopinExample + markAllCheckboxes);
+            const checkboxLength = getElementArrayLength(tablePopinExample + tableRow);
+            for (let i = 20; i < checkboxLength; i++) {
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/table.e2e-spec.ts:149]: why 20? not obvious for debugging/reading. Maybe you can make a var with a name that will make it easy to understand?

File: e2e/wdio/core/tests/table.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+            scrollIntoView(tableToolbarExample);
+            click(tableToolbarExample + button, 2);
+            for (let i = 0; i < 3; i++) {
+                setValue(dialogContent + inputField, 'test', i);
+            }
+            click(dialogContent + button, 1);
+            saveElementScreenshot(tableToolbarExample + table, 'table-toolbar-example-' + getImageTagBrowserPlatform(),
+                tablePage.getScreenshotFolder());
+            expect(checkElementScreenshot(tableToolbarExample + table, 'table-toolbar-example-' + getImageTagBrowserPlatform(),
+                tablePage.getScreenshotFolder())).toBeLessThan(5, `element item state mismatch`);
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/table.e2e-spec.ts:70]: maybe better to get initial row count, add new item, get new row count and check last row has the provided data?
screenshot is ok, but not many people are able to update them.

**Review by Seamoo13 — CHANGES_REQUESTED** [review]
please add additional cases:
- clear search
- toggle loading
- search in custom table config
- sort in custom table config
- cancel btn in custom table config
- table sort
- table filter

also please go through recent bugs related to table core and check that your tests cover them (if possible to cover)

**Review by maxym-valor — APPROVED** [review]

**Review by Seamoo13 — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**